src/gluonts/mx/block/quantile_output.py

```diff
@@ -183,103 +183,104 @@ def compute_quantile_weights(self) -> List:
         return quantile_weights
 
 
-class ProjectParams(nn.HybridBlock):
+class QuantileOutput:
     """
-    Defines a dense layer to compute the projection weights into the quantile
-    space.
+    Output layer using a quantile loss and projection layer to connect the
+    quantile output to the network.
 
     Parameters
     ----------
-    num_quantiles
-        number of quantiles to compute the projection.
-    is_iqf
-        determines whether to use IQF or QF.
+    quantiles
+        list of quantiles to compute loss over.
+
+    quantile_weights
+        weights of the quantiles.
     """
 
     @validated()
-    def __init__(self, num_quantiles: int, is_iqf: bool = False, **kwargs):
-        super().__init__(**kwargs)
+    def __init__(
+        self,
+        quantiles: List[float],
+        quantile_weights: Optional[List[float]] = None,
+    ) -> None:
+        self.quantiles = quantiles
+        self.num_quantiles = len(self.quantiles)
+        self.quantile_weights = quantile_weights
 
-        self.num_quantiles = num_quantiles
-        self.is_iqf = is_iqf
-        with self.name_scope():
-            if self.is_iqf:
-                self.proj_intrcpt = nn.Dense(1, flatten=False)
-                if self.num_quantiles > 1:
-                    self.proj_incrmnt = nn.Dense(
-                        self.num_quantiles - 1,
-                        flatten=False,
-                        activation="relu",
-                    )  # increments between quantile estimates
-            else:
-                self.projection = nn.Dense(
-                    units=self.num_quantiles, flatten=False
-                )
+    def get_loss(self) -> nn.HybridBlock:
+        return QuantileLoss(
+            quantiles=self.quantiles,
+            quantile_weights=self.quantile_weights,
+            is_equal_weights=True,
+        )
 
-    # noinspection PyMethodOverriding,PyPep8Naming
-    def hybrid_forward(self, F, x: Tensor) -> Tuple[Tensor, Tensor, Tensor]:
-        """
+    def get_quantile_proj(self, **kwargs) -> nn.HybridBlock:
```

Comment:
Does this class still need the `hybrid_forward` returning `self.projection(x)` on line 250?

Reply:
I believe that special case is still needed, because `self.proj_incrmnt` won't even be assigned as attribute in the constructor whenever `num_outputs == 1`

Reply:
yes